crates/weaver_resolver/src/registry.rs

```diff
@@ -512,6 +528,192 @@ fn resolve_include_constraints(ureg: &mut UnresolvedRegistry) -> bool {
 }
 
+fn resolve_inheritance_attrs(
+    attrs_group: &[UnresolvedAttribute],
+    attrs_parent_group: &[UnresolvedAttribute],
+    _group_lineage: Option<&mut GroupLineage>, // ToDo compute the lineage
+) -> Vec<UnresolvedAttribute> {
+    // A map attribute_id -> attribute_spec.
+    //
+    // Note: we use a BTreeMap to ensure that the resolved attributes are
+    // sorted by their id in the resolved registry. This is useful for unit
+    // tests to ensure that the resolved registry is easy to compare.
+    let mut resolved_attrs = BTreeMap::new();
+
+    // Inherit the attributes from the parent group.
+    for parent_attr in attrs_parent_group.iter() {
+        match &parent_attr.spec {
+            AttributeSpec::Ref { r#ref, .. } => {
+                _ = resolved_attrs.insert(r#ref.clone(), parent_attr.spec.clone())
+            }
+            AttributeSpec::Id { id, .. } => {
+                _ = resolved_attrs.insert(id.clone(), parent_attr.spec.clone())
+            }
+        }
+    }
+
+    // Override the inherited attributes with the attributes from the group.
+    for attr in attrs_group.iter() {
+        match &attr.spec {
+            AttributeSpec::Ref { r#ref, .. } => {
+                if let Some(parent_attr) = resolved_attrs.get(r#ref) {
+                    _ = resolved_attrs.insert(
+                        r#ref.clone(),
+                        resolve_inheritance_attr(&attr.spec, parent_attr),
+                    )
+                } else {
+                    _ = resolved_attrs.insert(r#ref.clone(), attr.spec.clone())
+                }
+            }
+            AttributeSpec::Id { id, .. } => {
+                _ = resolved_attrs.insert(id.clone(), attr.spec.clone())
+            }
+        }
+    }
+
+    resolved_attrs
+        .into_values()
+        .map(|spec| UnresolvedAttribute { spec })
+        .collect()
+}
+
+fn resolve_inheritance_attr(attr: &AttributeSpec, parent_attr: &AttributeSpec) -> AttributeSpec {
+    match attr {
+        AttributeSpec::Ref {
+            r#ref,
+            brief,
+            examples,
+            tag,
+            requirement_level,
+            sampling_relevant,
+            note,
+            stability,
+            deprecated,
+        } => {
+            match parent_attr {
+                AttributeSpec::Ref {
+                    brief: parent_brief,
+                    examples: parent_examples,
+                    tag: parent_tag,
+                    requirement_level: parent_requirement_level,
+                    sampling_relevant: parent_sampling_relevant,
+                    note: parent_note,
+                    stability: parent_stability,
+                    deprecated: parent_deprecated,
+                    ..
+                } => {
+                    // attr and attr_parent are both references.
+                    AttributeSpec::Ref {
+                        r#ref: r#ref.clone(),
+                        brief: if brief.is_some() {
```

Comment:
For all these options, you can make a helper method to clean this up.

```rust
fn clone_first_some<T : Clone>(first: &Option<T>, second: &Option<T>): Option<T> {
  if first.is_some() {
      first.clone()
  } else {
     second.clone()
  }    
```


Then all this code becomes:

```rust
AttribtueSpec::Ref {
   ...
  brief: clone_first_some(brief, parent_brief),
  examples: clone_first_some(examples, parent_examples),
  ...
}
```

Reply:
Created the following GH issue https://github.com/open-telemetry/weaver/issues/54